### PR TITLE
Fix tutorial for Qualcomm AI Engine Direct Backend

### PR DESCRIPTION
**Pull request by kirklandsign**
We have refactors recently and need to update the tutorial and cmake.

See https://github.com/pytorch/executorch/issues/2955 for isseues.

### CODE REVIEW

File: backends/qualcomm/CMakeLists.txt [hunk 1]
```diff
@@ -253,7 +253,7 @@ target_link_libraries(qnn_executorch_backend
     qnn_executorch_header
     qnn_schema
     qnn_manager
-    executorch
+    executorch_no_prim_ops
```
Comment (cccclai) [inline backends/qualcomm/CMakeLists.txt:256]: how do we link it with prim ops lib?

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md [hunk 2]
```diff
@@ -134,12 +138,12 @@ Commands to build `qnn_executor_runner` for Android:
 
 ```bash
 cd $EXECUTORCH_ROOT
-mkdir build_android
-cd build_android
+mkdir cmake_android_out
+cd cmake_android_out
```
Comment (cccclai) [inline docs/source/build-run-qualcomm-ai-engine-direct-backend.md:142]: why this change is needed?

Comment (kirklandsign) [reply docs/source/build-run-qualcomm-ai-engine-direct-backend.md:142]: can use old one. simply it's already in gitignore.

File: backends/qualcomm/CMakeLists.txt (hunk 1 above)
Comment (kirklandsign) [reply backends/qualcomm/CMakeLists.txt:256]: In https://github.com/pytorch/executorch/blob/687425d1c17c99acdcee2c0617e84e93fc3e1c13/examples/qualcomm/CMakeLists.txt#L102-L108, maybe full_portable_ops_lib has it… Otherwise we need to add executorch explicitly

For other users, add executorch explicitly to binary helps

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md (hunk 2 above)
Comment (cccclai) [reply docs/source/build-run-qualcomm-ai-engine-direct-backend.md:142]: let's use the old one probably? Try to make the change minimum

Comment (kirklandsign) [reply docs/source/build-run-qualcomm-ai-engine-direct-backend.md]: Done

File: backends/qualcomm/CMakeLists.txt (hunk 1 above)
Comment (kirklandsign) [reply backends/qualcomm/CMakeLists.txt:256]: TLDR is for users who use `qnn_executorch_backend` as a dependency, it should also have `executorch` as a dependency now.

**Comment by cccclai** [comment]
@chiwwang can you help with tagging reviewers?

**Comment by chiwwang** [comment]
Hey @chuntl , may I know if you're able to give this a try?

**Comment by chuntl** [comment]
Rebase to mainline and do the "./install_requirements.sh“ meet the target missing issue:

image

It seems there're some WIP PRs to fix the CI issue and affect to test this PR.

**Comment by chuntl** [comment]
> > Rebase to mainline and do the "./install_requirements.sh“ meet the target missing issue:
> > image
> > It seems there're some WIP PRs to fix the CI issue and affect to test this PR.
> 
> Got it. Assuming this error will be fixed, if the content of this PR looks good?

Yes, It looks good to me. But it's better to double-check after the error fixed.

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md [hunk 3]
```diff
@@ -138,8 +142,8 @@ mkdir build_android
 cd build_android
 # build executorch & qnn_executorch_backend
 cmake .. \
-    -DBUCK2=buck2 \
     -DCMAKE_INSTALL_PREFIX=$PWD \
+    -DEXECUTORCH_BUILD_SDK=ON \
     -DEXECUTORCH_BUILD_QNN=ON \
```
[the remark this answers is not in the document]
Comment (chiwwang) [reply docs/source/build-run-qualcomm-ai-engine-direct-backend.md:147]: never mind! I saw `-DCMAKE_INSTALL_PREFIX=$PWD`
So this might be caused that the `CMAKE_INSTALL_PREFIX` is not propagated.

**Comment by kirklandsign** [comment]
> Rebase to mainline and do the "./install_requirements.sh“ meet the target missing issue:
> 
> image
> 
> It seems there're some WIP PRs to fix the CI issue and affect to test this PR.

Since we updated https://github.com/pytorch/executorch/blob/main/build/cmake_deps.toml, we need to `rm -rf pip-out`. We will note this in the doc.

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kirklandsign) [reply docs/source/build-run-qualcomm-ai-engine-direct-backend.md:147]: Is this issue good now?

**Comment by chuntl** [comment]
Although I can't add into the reviewer list, it looks good to me, thank you!

File: docs/source/build-run-qualcomm-ai-engine-direct-backend.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (chuntl) [reply docs/source/build-run-qualcomm-ai-engine-direct-backend.md:147]: > Is this issue good now?

Yes, it's good now.

**Review by chuntl — APPROVED** [review]

**Review by cccclai — APPROVED** [review]
